src/dev/exit_on_warning.js

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const ignore = ['MaxListenersExceededWarning'];
```

Comment:
I'm ignoring `MaxListenersExceededWarning` because one of our tests (`src/cli/cluster/cluster_manager.test.ts`) emitted this warning and I don't consider these a security risk.

We could consider "fixing" the test so it doesn't emit this warning, but it's not super important. However, if we did, it would probably improve the codebase as new warnings of this type would not sneak in as those are usually a result of poorly written code that can hide other bugs. So if it's easy to do we should consider it (I spent a few hours trying to fix it without any luck).